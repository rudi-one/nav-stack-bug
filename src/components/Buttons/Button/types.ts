import {PressableProps} from 'react-native';

export interface ButtonProps extends PressableProps {
  text: String;
  type?: 'normal' | 'dark' | 'transparent';
}
