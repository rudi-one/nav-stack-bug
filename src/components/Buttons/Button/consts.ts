import {moderateScale} from 'src/utils/helpers';

export const BORDER_RADIUS = moderateScale(30);

export const BORDER_WIDTH = moderateScale(1.5);

export const TRANSPARENT_BUTTON_SCALE = 1 - BORDER_WIDTH / BORDER_RADIUS;
