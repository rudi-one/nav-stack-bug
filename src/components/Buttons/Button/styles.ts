import {StyleSheet} from 'react-native';
import {fontSize, fontWeight} from 'src/styles/typography';
import {colors} from 'src/styles/colors';
import {horizontalScale, verticalScale} from 'src/utils/helpers';
import {BORDER_RADIUS, BORDER_WIDTH, TRANSPARENT_BUTTON_SCALE} from './consts';

export default StyleSheet.create({
  container: {
    width: '50%',
    backgroundColor: colors.white,
    borderRadius: BORDER_RADIUS,
    marginHorizontal: horizontalScale(2),
    alignSelf: 'center',
    overflow: 'hidden',
  },
  pressable: {
    borderRadius: BORDER_RADIUS,
    alignItems: 'center',
    paddingVertical: verticalScale(15),
    paddingHorizontal: horizontalScale(10),
  },
  text: {
    color: colors.darkBlue,
    textAlign: 'center',
    ...fontWeight.bold,
    ...fontSize.x10,
  },
  disabled: {
    opacity: 0.9,
    backgroundColor: colors.darkMagenta,
  },
  transparentContainer: {
    borderColor: colors.white,
    borderWidth: BORDER_WIDTH,
    backgroundColor: colors.transparent,
    transform: [{scaleY: TRANSPARENT_BUTTON_SCALE}],
  },
  darkContainer: {
    backgroundColor: colors.darkBlue,
  },
  whiteText: {
    color: colors.white,
  },
});
