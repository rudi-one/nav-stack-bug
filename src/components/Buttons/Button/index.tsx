import React from 'react';
import {Pressable, Text, View} from 'react-native';
import {roundRippleConfig} from 'src/styles/consts';
import styles from './styles';
import {ButtonProps} from './types';

const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  const {disabled, type} = props;
  const isTransparent = type === 'transparent';
  const isDark = type === 'dark';
  const isWhiteText = isTransparent || isDark;

  return (
    <View
      style={[
        styles.container,
        disabled ? styles.disabled : {},
        isTransparent ? styles.transparentContainer : {},
        isDark ? styles.darkContainer : {},
      ]}>
      <Pressable
        {...props}
        style={styles.pressable}
        android_disableSound={true}
        android_ripple={roundRippleConfig}>
        <Text
          numberOfLines={1}
          style={[styles.text, isWhiteText ? styles.whiteText : {}]}>
          {props.text}
        </Text>
      </Pressable>
    </View>
  );
};

export default Button;
