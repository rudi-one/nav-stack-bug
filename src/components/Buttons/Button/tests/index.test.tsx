import {render} from '@testing-library/react-native';
import Button from 'buttons/Button';
import React from 'react';

describe('<Button /> tests', () => {
  it('should match normal button snapshot', () => {
    const component = <Button text={'test'} type={'normal'} />;

    expect(render(component)).toMatchSnapshot();
  });
  it('should match normal disabled button snapshot', () => {
    const component = <Button text={'test'} type={'normal'} disabled={true} />;

    expect(render(component)).toMatchSnapshot();
  });
  it('should match dark button snapshot', () => {
    const component = <Button text={'test'} type={'dark'} />;

    expect(render(component)).toMatchSnapshot();
  });
  it('should match dark disabled button snapshot', () => {
    const component = <Button text={'test'} type={'dark'} disabled={true} />;

    expect(render(component)).toMatchSnapshot();
  });
  it('should match transparent button snapshot', () => {
    const component = <Button text={'test'} type={'transparent'} />;

    expect(render(component)).toMatchSnapshot();
  });
  it('should match transparent disabled button snapshot', () => {
    const component = (
      <Button text={'test'} type={'transparent'} disabled={true} />
    );

    expect(render(component)).toMatchSnapshot();
  });
});
