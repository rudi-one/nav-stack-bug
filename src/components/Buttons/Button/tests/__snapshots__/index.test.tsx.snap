// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should match dark button snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 82,
        "marginHorizontal": 4.5,
        "overflow": "hidden",
        "width": "50%",
      },
      Object {},
      Object {},
      Object {
        "backgroundColor": "#292C5A",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 82,
        "paddingHorizontal": 21.5,
        "paddingVertical": 29.5,
      }
    }
    text="test"
    type="dark"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`<Button /> tests should match dark disabled button snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 82,
        "marginHorizontal": 4.5,
        "overflow": "hidden",
        "width": "50%",
      },
      Object {
        "backgroundColor": "#a8006c",
        "opacity": 0.9,
      },
      Object {},
      Object {
        "backgroundColor": "#292C5A",
      },
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 82,
        "paddingHorizontal": 21.5,
        "paddingVertical": 29.5,
      }
    }
    text="test"
    type="dark"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`<Button /> tests should match normal button snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 82,
        "marginHorizontal": 4.5,
        "overflow": "hidden",
        "width": "50%",
      },
      Object {},
      Object {},
      Object {},
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 82,
        "paddingHorizontal": 21.5,
        "paddingVertical": 29.5,
      }
    }
    text="test"
    type="normal"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`<Button /> tests should match normal disabled button snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 82,
        "marginHorizontal": 4.5,
        "overflow": "hidden",
        "width": "50%",
      },
      Object {
        "backgroundColor": "#a8006c",
        "opacity": 0.9,
      },
      Object {},
      Object {},
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 82,
        "paddingHorizontal": 21.5,
        "paddingVertical": 29.5,
      }
    }
    text="test"
    type="normal"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`<Button /> tests should match transparent button snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 82,
        "marginHorizontal": 4.5,
        "overflow": "hidden",
        "width": "50%",
      },
      Object {},
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#FFFFFF",
        "borderWidth": 4,
        "transform": Array [
          Object {
            "scaleY": 0.9512195121951219,
          },
        ],
      },
      Object {},
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 82,
        "paddingHorizontal": 21.5,
        "paddingVertical": 29.5,
      }
    }
    text="test"
    type="transparent"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`<Button /> tests should match transparent disabled button snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 82,
        "marginHorizontal": 4.5,
        "overflow": "hidden",
        "width": "50%",
      },
      Object {
        "backgroundColor": "#a8006c",
        "opacity": 0.9,
      },
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#FFFFFF",
        "borderWidth": 4,
        "transform": Array [
          Object {
            "scaleY": 0.9512195121951219,
          },
        ],
      },
      Object {},
    ]
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 82,
        "paddingHorizontal": 21.5,
        "paddingVertical": 29.5,
      }
    }
    text="test"
    type="transparent"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#292C5A",
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
